comlink.ts

```diff
@@ -26,6 +26,17 @@ export interface Endpoint {
     options?: {}
   ): void;
 }
+type ProxyResult<T> =
+  // T is a class
+  T extends { new (...args: infer Arguments): infer R } ? { new (...args: Arguments): Promise<R> } :
```

Comment:
In my understanding, the methods of `R` return promises.

Reply:
you're right. thanks!